addons/delivery/models/delivery_carrier.py

```diff
@@ -423,112 +357,3 @@ def _get_price_from_picking(self, total, weight, volume, quantity):
             raise UserError(_("Not available for current order"))
 
         return price
-
-    def base_on_rule_send_shipping(self, pickings):
-        res = []
-        for p in pickings:
-            carrier = self._match_address(p.partner_id)
-            if not carrier:
-                raise ValidationError(_('There is no matching delivery rule.'))
-            res = res + [{'exact_price': p.carrier_id._get_price_available(p.sale_id) if p.sale_id else 0.0,  # TODO cleanme
-                          'tracking_number': False}]
-        return res
-
-    def base_on_rule_get_tracking_link(self, picking):
-        return False
-
-    def base_on_rule_cancel_shipment(self, pickings):
-        raise NotImplementedError()
-
-    # -------------------------------- #
-    # get default packages/commodities #
-    # -------------------------------- #
-
-    def _get_packages_from_order(self, order, default_package_type):
```

Comment:
Well spotted... There is definitely something to correct/improve here.
At first, I moved all weight-related stuff to stock_delivery, but it led to several overrides of methods. Since weight is a standard field from product.product, it has been decided that it should remain a variable for the price rule.
I'll discuss that matter tomorrow and see what is best to be done in the current PR.

Reply:
Current idea to fix this: add the same group on the wizard div showing the weight (and removing the group if stock installed). So we show the weight only if we can edit it.
There surely is room for improvements, but should do the job for now.